### ISSUE TITLE
variable-name doesn't work on functions

### ISSUE THREAD

**Issue by wolfy1339** [palantir/tslint#4375]
### Bug Report

- __TSLint version__:  5.11.0
- __TypeScript version__: 3.2.2
- __Running TSLint via__: VSCode

#### TypeScript code being linted

```ts
/**
 * @param  {string} s The commit hash string
 * @return {string}   Returns formatted commit hash string
 */
function fmt_url(s: string): string {
  return `\x0302\x1F${s}\x0F`;
}

```

with `tslint.json` configuration:

```json
{
    "defaultSeverity": "error",
    "extends": [
        "tslint:recommended",
    ],
    "jsRules": {},
    "rules": {
        "variable-name": [
            true,
            "allow-leading-underscore",
            "allow-pascal-case"
        ]
    }
}
```

#### Actual behavior
Doesn't complain at all for the function name

#### Expected behavior
To lint all functions, classes and anything that is assigned a name

**Comment by JoshuaKGoldberg** [comment]
@wolfy1339 you're in luck! [`tslint-microsoft-contrib`'s function-name](https://github.com/microsoft/tslint-microsoft-contrib) rule does this.

See #1142 for discussion on which rules will be imported into TSLint core.
